src/poll.rs

```diff
@@ -145,3 +270,339 @@ pub fn selector(poll: &Poll) -> &sys::Selector {
 pub fn selector_mut(poll: &mut Poll) -> &mut sys::Selector {
     &mut poll.selector
 }
+
+/*
+ *
+ * ===== Registration =====
+ *
+ */
+
+impl Registration {
+    pub fn new(poll: &Poll, token: Token, interest: EventSet, opts: PollOpt) -> Registration {
+        let queue = poll.readiness_queue.clone();
+        let node = queue.new_readiness_node(token, interest, opts);
+
+        Registration {
+            node: node,
+            queue: queue,
+        }
+    }
+
+    pub fn update(&self, poll: &Poll, interest: EventSet, opts: PollOpt) -> io::Result<()> {
+        if !self.queue.identical(&poll.readiness_queue) {
+            return Err(io::Error::new(io::ErrorKind::Other, "nope"));
+        }
+
+        self.node().interest.set(interest);
```

Comment:
This is kinda surprising to me in terms of what's going on. We're modifying a `Cell` which is explicitly documented as only accessible from `Poll`, which sounds like a data race waiting to happen?

This is sort of what I was alluding to above with where the unsafety here may be a little too much to reason about. It's very unclear which operations are unsafe and which aren't because the `unsafe` blocks are so wildly powerful in this module.